arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1712 @@
+			"post": {
+				"tags": ["POST"],
+				"summary": "Import all the Devices in an IotHub",
+				"description": "Import all the Devices in an IotHub",
+				"operationId": "IotHubResource_ImportDevices",
+				"consumes": ["application/json", "text/json"],
+				"produces": ["application/json", "text/json"],
+				"parameters": [{
+					"name": "resourceName",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "importDevicesRequest",
+					"in": "body",
+					"required": true,
+					"schema": {
+						"$ref": "#/definitions/ImportDevicesRequest"
+					}
+				}, {
+					"name": "subscriptionId",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"name": "resourceGroupName",
+					"in": "path",
+					"required": true,
+					"type": "string"
+				}, {
+					"$ref": "#/parameters/api-version"
+				}],
+				"responses": {
+					"200": {
+						"description": "OK",
+						"schema": {
+							"$ref": "#/definitions/JobResponse"
+						}
+					},
+					"default": {
+						"description": "DefaultErrorResponse",
+						"schema": {
+							"$ref": "#/definitions/ErrorDetails"
+						}
+					}
+				},
+				"deprecated": false
+			}
+		}
+	},
+	"definitions": {
+		"Object": {
+			"type": "object",
+			"properties": {}
+		},
+		"SharedAccessSignatureAuthorizationRule": {
+			"type": "object",
+			"properties": {
+				"keyName": {
+					"type": "string"
+				},
+				"primaryKey": {
+					"type": "string"
+				},
+				"secondaryKey": {
+					"type": "string"
+				},
+				"rights": {
+					"enum": ["RegistryRead", "RegistryWrite", "ServiceConnect", "DeviceConnect"],
+					"type": "string"
+				}
+			}
+		},
+		"IotHubProperties": {
+			"type": "object",
+			"properties": {
+				"authorizationPolicies": {
+					"type": "array",
+					"items": {
+						"$ref": "#/definitions/SharedAccessSignatureAuthorizationRule"
+					}
+				},
+				"hostName": {
+					"type": "string"
+				},
+				"eventHubEndpoints": {
+					"type": "object",
+					"additionalProperties": {
+						"$ref": "#/definitions/EventHubProperties"
+					}
+				},
+				"storageEndpoints": {
+					"type": "object",
+					"additionalProperties": {
+						"$ref": "#/definitions/StorageEndpointProperties"
+					}
+				},
+				"messagingEndpoints": {
+					"type": "object",
+					"additionalProperties": {
+						"$ref": "#/definitions/MessagingEndpointProperties"
+					}
+				},
+				"enableFileUploadNotifications": {
+					"type": "boolean"
+				},
+				"cloudToDevice": {
+					"$ref": "#/definitions/CloudToDeviceProperties"
+				},
+				"comments": {
+					"type": "string"
+				},
+				"operationsMonitoringProperties": {
+					"$ref": "#/definitions/OperationsMonitoringProperties"
+				},
+				"features": {
+					"enum": ["None", "DeviceManagement"],
+					"type": "string"
+				}
+			}
+		},
+		"IotHubSkuInfo": {
+			"type": "object",
+			"properties": {
+				"name": {
+					"enum": ["F1", "S1", "S2"],
```

Comment:
It will definitely be a string. However, if you think that is a finite set and wouldn't be expanded in the near future then modelAsString: false will direct Autorest to generate it as an enum in languages like C#, java, etc. that support an enum. For javascript the generated code will do case insensitive validation against the set of allowed values.

If you think that you will be adding more values to the enum in the near future then modelAsString: true. This will direct AutoRest to generate a static class with constant properties in strongly typed languages. Thus customers can still get intellisense.